obs-studio-client/source/controller.cpp

```diff
@@ -237,80 +282,23 @@ std::shared_ptr<ipc::client> Controller::host(std::string uri) {
 	if (m_isServer)
 		return nullptr;
 
-	// Store info
-	std::string program = serverBinaryPath;
 	std::string commandLine = '"' + serverBinaryPath + '"' + " " + uri;
-	std::string workingDirectory = serverWorkingPath.length() > 0 ? serverWorkingPath : get_working_directory();
-
-	// Test for existing process.
-	std::string pid_path;
-
-#ifdef _WIN32
-
-	std::basic_string<TCHAR> tmp;
-	tmp.resize(MAX_PATH + 1);
-
-	DWORD tmp_len = GetTempPath(MAX_PATH + 1, &tmp[0]);
-
-#ifdef UNICODE
-	{
 
-	DWORD long_path_len = GetLongPathName(&tmp[0], NULL, 0);
-	std::basic_string<TCHAR> long_tmp;
-	long_tmp.resize(long_path_len);
-	GetLongPathName(&tmp[0], &long_tmp[0], long_path_len);
+	std::string workingDirectory =
+		serverWorkingPath.empty() ? get_working_directory() : serverWorkingPath;
 
-	pid_path.assign(from_utf16_wide_to_utf8(long_tmp.data(),long_path_len));
+	// Test for existing process.
+	std::string pid_path(get_temp_directory());
+	pid_path.append("server.pid");
```

Comment:
Is get_temp_directory() guaranteed to return a path with / (or \) at the end?

Reply:
Yeah, so contrary to MSDN's docs, where it gives the example "C:\TEMP", it would actually return "C:\TEMP\".